### PR TITLE
[CP-18781] Always se the correct `domid` value in `update_vif`

### PR DESCRIPTION
**Pull request by mseri**
When updating the MTU of a VIF it can happen that the VM domid is still out of sync. We can use the correct `domid` value passing it via `xenopsd` as a field in `VIF.stat`.

See also  https://github.com/xapi-project/xcp-idl/pull/131 and  https://github.com/xapi-project/xenopsd/pull/288
Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

### CODE REVIEW

File: ocaml/xapi/xapi_xenops.ml [hunk 1]
```diff
@@ -1639,10 +1639,21 @@ let update_vif ~__context id =
...
-									let device = "vif" ^ (Int64.to_string (Db.VM.get_domid ~__context ~self:vm)) ^ "." ^ (snd id) in
-									let dbg = Context.string_of_task __context in
-									let mtu = Net.Interface.get_mtu dbg ~name:device in
-									Db.VIF.set_MTU ~__context ~self:vif ~value:(Int64.of_int mtu)
+										(* We need to use the value of domid in VIF.stat because *)
+										(* this function could have been called before the information *)
+										(* in the database is completely updated *)
+										let some_domid = Opt.map (fun x -> (snd x).domid) info
+												|> Opt.join in
+										match some_domid with
```
Comment (robhoes) [inline ocaml/xapi/xapi_xenops.ml]: We are already inside a match on `info`, where `state` is the record you need. So you can just match on `state.domid` here. Or `state.device` when we switch from domid to device name in the type (see comment on the xcp-idl PR).

Comment (robhoes) [inline ocaml/xapi/xapi_xenops.ml]: Also, the indentation looks one step too far to the right.

File: ocaml/xapi/xapi_xenops.ml [hunk 2]
```diff
@@ -1639,10 +1639,21 @@ let update_vif ~__context id =
...
+											| Some domid -> begin
```
Comment (robhoes) [inline ocaml/xapi/xapi_xenops.ml]: You don't need begin/end here.

File: ocaml/xapi/xapi_xenops.ml [hunk 3]
```diff
@@ -1639,10 +1639,21 @@ let update_vif ~__context id =
...
+													let device = "vif" ^ (string_of_int domid) ^ "." ^ (snd id) in
+													let dbg = Context.string_of_task __context in
+													let mtu = Net.Interface.get_mtu dbg ~name:device in
+													Db.VIF.set_MTU ~__context ~self:vif ~value:(Int64.of_int mtu)
+											end
+											| None ->
+													debug "could not determine domid for VIF %s.%s" (fst id) (snd id);
```
Comment (robhoes) [inline ocaml/xapi/xapi_xenops.ml]: This log line will now appear twice.

File: ocaml/xapi/xapi_xenops.ml [hunk 4]
```diff
@@ -1639,10 +1639,21 @@ let update_vif ~__context id =
...
+													raise (Failure "domid not found");
```
Comment (robhoes) [inline ocaml/xapi/xapi_xenops.ml]: You can use `failwith` instead.

**Comment by mseri** [comment]
I make a new PR with the latest changes, based on the latest master.

**Pull request closed without merging**